src/backend/commands/createas.c

```diff
@@ -609,28 +419,239 @@ ExecCreateTableAs(CreateTableAsStmt *stmt, const char *queryString,
 
 		if (into->ivm)
 		{
-
 			Oid matviewOid = address.objectId;
 			Relation matviewRel = table_open(matviewOid, NoLock);
 			Relids	relids = NULL;
 
-			copied_query = copyObject(query);
-			AcquireRewriteLocks(copied_query, true, false);
+			/*
+			 * Mark relisivm field, if it's a matview and into->ivm is true.
+			 */
+			SetMatViewIVMState(matviewRel, true);
 
-			CreateIvmTriggersOnBaseTables(copied_query, (Node *)copied_query->jointree, matviewOid, &relids);
+			if (!into->skipData)
+			{
+				copied_query = copyObject(query);
+				AcquireRewriteLocks(copied_query, true, false);
 
+				CreateIvmTriggersOnBaseTables(copied_query, (Node *)copied_query->jointree, matviewOid, &relids);
+				bms_free(relids);
+			}
 			table_close(matviewRel, NoLock);
-
-			bms_free(relids);
 		}
 	}
 
 	return address;
 }
+
+/*
+ * rewriteQueryForIMMV -- rewrite query for definition query of IMMV
```

Comment:
"rewrite view definition query for IMMV" ?

Reply:
Thank you for your review. I will fix sentence as you say.